### ISSUE TITLE
Crash app when downloading video

### ISSUE THREAD

**Issue by Amirraan** [libre-tube/LibreTube#3181]
### Steps to reproduce

When downloading the movie, if we stop downloading and come out of the app, and we go back to the app, the app will crash and I can't continue downloading

One suggestion:
Please mention the approximate volume of the movie before starting downloading the movie in different quality 

### Expected behavior

java.lang.IllegalArgumentException: Service not registered: com.github.libretube.ui.fragments.DownloadsFragment$serviceConnection$1@63966d6
 at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1987)
 at android.app.ContextImpl.unbindService(ContextImpl.java:1904)
 at android.content.ContextWrapper.unbindService(ContextWrapper.java:741)
 at android.content.ContextWrapper.unbindService(ContextWrapper.java:741)
 at com.github.libretube.ui.fragments.DownloadsFragment.onStop(DownloadsFragment.kt:16)
 at androidx.fragment.app.FragmentStateManager.stop(FragmentStateManager.java:65)
 at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:178)
 at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1029)
 at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:92)
 at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:74)
 at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:4)
 at android.os.Handler.handleCallback(Handler.java:900)
 at android.os.Handler.dispatchMessage(Handler.java:103)
 at android.os.Looper.loop(Looper.java:219)
 at android.app.ActivityThread.main(ActivityThread.java:8668)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)

### Actual behavior

.. 

### LibreTube version

0.12.0

### Android version

10

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Amirraan** [comment]
Ok, thanks

**Comment by ThomasLeigh** [comment]
I've observed this issue also in regard to finished downloads. Once a video is downloaded (either as video or just audio), the app's window disappears - once I open it again it shows the following error:

java.lang.RuntimeException: Error receiving broadcast Intent { act=com.github.libretube.services.DownloadService.ACTION_SERVICE_STOPPED flg=0x10 } in com.github.libretube.receivers.DownloadReceiver@6e391d3
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1568)
	at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:224)
	at android.app.ActivityThread.main(ActivityThread.java:7590)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
Caused by: android.content.ActivityNotFoundException: Unable to find explicit activity class {com.github.libretube/com.github.libretube.ui.activities.MainActivity}; have you declared this activity in your AndroidManifest.xml?
	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2064)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1720)
	at android.app.Activity.startActivityForResult(Activity.java:5258)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:2)
	at android.app.Activity.startActivityForResult(Activity.java:5216)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:1)
	at android.app.Activity.startActivity(Activity.java:5587)
	at android.app.Activity.startActivity(Activity.java:5555)
	at com.github.libretube.receivers.DownloadReceiver.onReceive(DownloadReceiver.kt:64)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1558)
	... 8 more
    

LibreTube 0.12.1
Android 10
MIUI 12.5.1

**Comment by Amirraan** [comment]
I'm still using version 0.12.0 and I'm waiting for version 0.13.0 to be
released
